components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,76 @@
+const reddit = require("../../reddit.app.js");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emits an event each time a new link is added to a subreddit",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    reddit,
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "The subreddit you'd like to watch for new links.",
+      default: "redditdev",
+    },
+    timer: {
+      label: "Polling schedule",
+      description: "Pipedream polls Reddit for new links on this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minutes.
+      },
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emits 10 sample events on the first run during deploy.
+      let before = null;
+
+      const reddit_things = await this.reddit.getNewSubredditLinks(
+        null,
+        this.subreddit,
+        10
+      );
+
+      const links_pulled = this.reddit.wereLinksPulled(reddit_things);
+      if (links_pulled) {
+        before = reddit_things.data.children[0].data.name;
+        reddit_things.data.children.forEach((reddit_link) => {
+          this.emitRedditEvent(reddit_link);
+        });
+      }
+
+      this.db.set("before", before);
+    },
+  },
+  methods: {
+    emitRedditEvent(reddit_event) {
+      const { name: id, title: summary } = reddit_event.data;
+      this.$emit(reddit_event, {
+        id,
+        summary,
+      });
+    },
+  },
+  async run() {
+    const start_before = this.db.get("before");
+    do {
+      const reddit_things = await this.reddit.getNewSubredditLinks(
+        start_before,
```

Comment:
Since you pass `start_before` every time, and don't change that value anywhere in this loop, the loop never terminates (and you're not paginating correctly). You want to set `start_before` to the value of `new_before` so that you pass the new `before` value to the `getNewSubredditLinks ` method, correct?

Reply:
Good catch!

So that error is due to my thought process when addressing pagination -- at first I thought I would need to variables observing the error value one for updating onto the db prop and the other to exit the loop. At the end only one was needed. THanks